src/job.rs

```diff
@@ -127,31 +144,62 @@ impl Job {
                 key: hasher.finish(),
                 phantom: PhantomData,
             },
-            env: unwrapped.env,
-            command: unwrapped.command,
+            env: &unwrapped.env,
+            command: &unwrapped.command,
             input_files,
+            input_jobs,
             outputs,
         })
     }
+
+    pub fn final_key(
+        &self,
+        path_to_hash: &HashMap<PathBuf, String>,
+        job_to_content_hash: &HashMap<Key<Base>, store::Item>,
+    ) -> Result<Key<Final>> {
+        let mut hasher = Xxh3::new();
+
+        self.base_key.hash(&mut hasher);
+
+        for path in &self.input_files {
+            match path_to_hash.get(path) {
+                Some(hash) => {
+                    path.hash(&mut hasher);
```

Comment:
This reads a bit weird given it seems to be rehashing a hash and also hashing a path that we already have the hash for. I assume this is just a naming confusion where path_to_hash is return the hash of the file located at the path?

Reply:
mm, good point on the path already being hashed. That's easier to see now that it all lives in the same `impl`.

I'm not 100% sure what the last sentence means here but `path_to_hash` is what you described (a mapping from path to hash of the file located at that path.)